src/test/java/com/mongodb/jdbc/MongoSQLStatementTest.java

```diff
@@ -62,75 +71,71 @@ void testNoop(MongoSQLConnectionTest.TestInterface ti) throws SQLException {
+
+        when(mongoCursor.next())
+                .thenAnswer(
+                        invocation -> {
+                            rowCnt.incrementAndGet();
+                            return generateRow();
+                        });
+
+        when(mongoDatabase.runCommand(any(), eq(MongoJsonSchemaResult.class)))
+                .thenReturn(generateSchema());
+
+        ResultSet rs = mongoStatement.executeQuery("select * from foo");
+        ResultSetMetaData metaData = rs.getMetaData();
+        assertEquals(9, metaData.getColumnCount());
+
+        rs.next();
+        assertThrows(
+                SQLException.class,
+                () -> {
+                    rs.getInt(1);
+                });
+        assertFalse(rs.next());
+        assertThrows(
+                SQLException.class,
+                () -> {
+                    rs.getInt(1);
+                });
+
+        assertTrue(rs.isLast());
     }
 
     @Test
     void testExecuteQuery() throws SQLException {
-        // TODO: SQL-535, requires MongoResultSet implementation
-        //        AtomicInteger rowCnt = new AtomicInteger();
-        //        when(mongoCursor.hasNext()).thenAnswer(invocation -> rowCnt.get() < 2);
-        //
-        //        when(mongoCursor.next())
-        //                .thenAnswer(
-        //                        invocation -> {
-        //                            if (rowCnt.incrementAndGet() == 1) {
-        //                                return generateMetadataDoc();
-        //                            }
-        //                            return generateRow();
-        //                        });
-        //
-        //        ResultSet rs = mongoStatement.executeQuery("select * from foo");
-        //        ResultSetMetaData metaData = rs.getMetaData();
-        //        assertEquals(2, metaData.getColumnCount());
-        //        // need to call next() first
-        //        assertThrows(
-        //                SQLException.class,
-        //                new Executable() {
-        //                    @Override
-        //                    public void execute() throws Throwable {
-        //                        rs.getInt(1);
-        //                    }
-        //                });
-        //        assertTrue(rs.next());
-        //        assertEquals(1, rs.getInt(1));
-        //        assertEquals("test", rs.getString(2));
-        //        assertFalse(rs.next());
-        //        assertTrue(rs.isLast());
+        AtomicInteger rowCnt = new AtomicInteger();
+        when(mongoDatabase.runCommand(any(), eq(MongoJsonSchemaResult.class)))
+                .thenReturn(generateSchema());
+        when(mongoCursor.hasNext()).thenAnswer(invocation -> rowCnt.get() < 1);
+
+        when(mongoCursor.next())
+                .thenAnswer(
+                        invocation -> {
+                            if (rowCnt.incrementAndGet() == 1) {
```

Comment:
I am not sure I follow what is going on here. The if will increment the row count, then if rowCnt == 1 it will call generateRow() and if not, it will also call generateRow. Is this a left over from the previous version of the test?

Reply:
I think this is because in the `MySQL` implementation we sent the `metadata` in the first document of the cursor, but now the metadata is coming from a separate call to ADL. I think this should be changed!

Reply:
That's right, it came from the `MySQL` test where the first one would return a `generateMetadataDoc()`, and was left in on accident and still worked since the behavior was the same.  I fixed it now.